submissions/recipe_oriented_factorio_life/recipe_oriented_factorio_life.md

```diff
@@ -0,0 +1,75 @@
+
+Next, let‘s look at an advanced circuit module with three types of raw materials. By setting the logistic train stop requesting multiple ingredients, we were able to unload all of our raw materials at a single station. At this point all the raw materials are mixed in one warehouse， we then use filter miniloaders (similar to filter inserters) to separate each material onto a separate belt to supply the assembly machine.
+
+![Advanced circuit module](figure/advanced_circuit_module.jpg)
+_Advanced circuit module. Use a single station to obtain raw materials._
+
+The feedstock of recipes that produce/consume fluids can also be transported in a fluid wagon, such as oil processing.
+
+![Oil refinery module](figure/oil_refinery_module.jpg)
+_Oil refinery module. Note that by controlling the pump via the LTN signal allows different fluids to be loaded or unloaded at the same station without mixing._
+
+There are also some unconventional recipes, such as research, thermal power plant, and rocket silo.
+
+![research center module](figure/research_center_module.jpg)
+_Research center module, which doesn't need an output station_
+
+![thermal power plant module](figure/thermal_power_plant_module.jpg)
+_Thermal power plant with landfill to place on water._
+
+![rocket silo module](figure/rocket_silo_module.jpg)
+_Rocket silo module. Independent logistics robot network can be used within the module. The bottom station is for materials needed to build rockets, and the station on right is for payloads of the rocket, usually satellites._
+
+### Pros and cons of ROFL
+
+ROFL is optimized towards engineers, making it easy for players to divide and conquer production problems. When you need to implement a certain recipe, simply set up input and output stations in the grid that match the types of raw materials and products, and then design production lines between the stations. When the capacity of a recipe is insufficient, you can directly copy and paste the corresponding factory module and integrate it into the whole factory production without [plumbing up the inputs and outputs](https://alt-f4.blog/ALTF4-22/#the-goal-of-factorio).
+
+ROFL is not optimized towards production, it is not a perfect-ratio production line, so it's common that structures stop working because of insufficient raw materials or product overflow. Each item has to be loaded to and unloaded from a wagon, adding a lot of input and output bottlenecks, most typically in copper wire, where the number of products exceeds the number of raw materials and the number of stacks is high while the demand is huge. There is a very significant cache with the warehouses, which is not conducive to identify the items with insufficient production in time (definitely iron ore though). The rail network used for inter-module logistics in the factory is expensive to build and also poses a large number of traffic accident hazards (praise the spidertron).
+
```

Comment:
Maybe add a section here about how you can use the concepts of ROFL in a less extreme way to divide and conquer complex recipes into more manageable chunks. This could make the piece more directly useful for newer players that may get overwhelmed by the complexity of some end game recipes.

Reply:
I wrote about extending ROFL in the Conclusion section, but it is quite abstract. I think it doesn't need more generalized examples, focus on explaining the modularization idea is enough.